tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.hh

```diff
@@ -0,0 +1,194 @@
+		bool compare_eso_stream( std::string const & expected_string, bool reset_stream = true ) {
+			return ioHelper->compare_eso_stream( expected_string, reset_stream );
+		}
+
+		// Compare an expected string against the MTR stream. The default is to reset the MTR stream after every call.
+		// It is easier to test successive functions if the MTR stream is 'empty' before the next call.
+		bool compare_mtr_stream( std::string const & expected_string, bool reset_stream = true ) {
+			return ioHelper->compare_mtr_stream( expected_string, reset_stream );
+		}
+
+		// Compare an expected string against the ECHO stream. The default is to reset the ECHO stream after every call.
+		// It is easier to test successive functions if the ECHO stream is 'empty' before the next call.
+		bool compare_echo_stream( std::string const & expected_string, bool reset_stream = true ) {
+			return ioHelper->compare_echo_stream( expected_string, reset_stream );
+		}
+
+		// Compare an expected string against the COUT stream. The default is to reset the COUT stream after every call.
+		// It is easier to test successive functions if the COUT stream is 'empty' before the next call.
+		bool compare_cout_stream( std::string const & expected_string, bool reset_stream = true ) {
+			return ioHelper->compare_cout_stream( expected_string, reset_stream );
+		}
+
+		// Compare an expected string against the CERR stream. The default is to reset the CERR stream after every call.
+		// It is easier to test successive functions if the CERR stream is 'empty' before the next call.
+		bool compare_cerr_stream( std::string const & expected_string, bool reset_stream = true ) {
+			return ioHelper->compare_cerr_stream( expected_string, reset_stream );
+		}
+
+		// Check if ESO stream has any output. Useful to make sure there are or are not outputs to ESO.
+		bool has_eso_output() {
+			return ioHelper->has_eso_output();
+		}
+
+		// Check if MTR stream has any output. Useful to make sure there are or are not outputs to MTR.
+		bool has_mtr_output() {
+			return ioHelper->has_mtr_output();
+		}
+
+		// Check if ECHO stream has any output. Useful to make sure there are or are not outputs to ECHO.
+		bool has_echo_output() {
+			return ioHelper->has_echo_output();
+		}
+
+		// Check if COUT stream has any output. Useful to make sure there are or are not outputs to COUT.
+		bool has_cout_output() {
+			return ioHelper->has_cout_output();
+		}
+
+		// Check if CERR stream has any output. Useful to make sure there are or are not outputs to CERR.
+		bool has_cerr_output() {
+			return ioHelper->has_cerr_output();
+		}
+
+		// This function processes an idf snippet and defaults to using the idd cache for the fixture.
+		// The cache should be used for nearly all calls to this function.
+		// This more or less replicates InputProcessor::ProcessInput() but in a more usable fashion for unit testing
+		bool process_idf( std::string const & idf_snippet, bool use_idd_cache = true ) {
```

Comment:
process_idf does not test InputProcessor. It uses the same functions in InputProcessor however. This is really just a helper function for unit testing only. I test (most) of InputProcessor in InputProcessor.unit.cc. This new process_idf() function wraps all needed InputProcessor functions in a much more unit testing friendly way, instead of just calling InputProcessor::ProcessInput(). This new function also allows for passing only IDF snippets and more importantly, passing them as strings instead of opening IDF files.